app/dao/notifications_dao.py

```diff
@@ -597,3 +601,20 @@ def notifications_not_yet_sent(should_be_sending_after_seconds, notification_typ
         Notification.status == NOTIFICATION_CREATED
     ).all()
     return notifications
+
+
+def guess_notification_type(search_term):
+    try:
+        return (
+            SMS_TYPE,
+            try_validate_and_format_phone_number(''.join(
+                str(int(character))
+                if character not in {'(', ')', ' ', '-', '+'} else ''
+                for character in search_term
+            )).lstrip('+0'),
+        )
```

Comment:
if we're already failing on letters, and stripping non numeric characters here, and we explicitly allow partials, what does `try_and_validate` give us?

Reply:
actually, would `search_for_email = set(search_term) & set(string.ascii_letters + '@')` catch the actual likely search terms well enough? if they search for something weird like `!"£$%^&*(` it would match a phone num, though i think that's fine as who knows what they mean then

Reply:
Yeah, that set magic works just as well. I’ve changed it to that, squashed it into ad423c4 and refactored a bit to not do the normalisation in two different places.